apps/testingaccessibility/src/context/progress-context.tsx

```diff
@@ -24,36 +26,37 @@ export function useProgress() {
 
 export const ProgressContext = React.createContext(defaultProgressContext)
 
-export const ProgressProvider: React.FC = ({children}) => {
-  const [isLoadingProgress, setIsLoadingProgress] =
-    React.useState<boolean>(true)
-  const [progress, setProgress] = React.useState<LessonProgress[]>([])
-  const {status} = useSession()
-
-  const fetchProgress = React.useCallback(async () => {
-    if (status === 'authenticated') {
-      const progress = await getLessonProgressForUser()
-      setProgress(progress)
-    }
+function useLoadProgress() {
+  const {status: sessionStatus} = useSession()
 
-    if (status !== 'loading') {
-      setIsLoadingProgress(false)
+  const {
+    data,
+    status: loadingProgressStatus,
+    refetch,
+  } = useQuery(['load progress', sessionStatus], async ({queryKey}) => {
+    if (last(queryKey) === 'authenticated') {
+      return await getLessonProgressForUser()
+    } else {
+      return []
     }
-  }, [status])
+  })
 
-  React.useEffect(() => {
-    if (status === 'authenticated') {
-      fetchProgress().catch(console.error)
-    }
-  }, [fetchProgress, status])
+  return {
+    progress: data,
+    status: loadingProgressStatus,
+    refetch,
+  }
+}
 
+export const ProgressProvider: React.FC = ({children}) => {
+  const {progress, status, refetch} = useLoadProgress()
   return (
     <ProgressContext.Provider
       value={{
         progress,
-        isLoadingProgress,
+        isLoadingProgress: status === 'loading',
         toggleLessonComplete: async (slug: string) => {
-          await toggleLessonProgressForUser({slug}).then(() => fetchProgress())
+          return await toggleLessonProgressForUser({slug}).finally(refetch)
```

Comment:
TODO: test if the refetch works as expected. 

@vojtaholik I noticed you'd added some options to the useQuery call. I think using the `status` prop in the queryKey should keep it refreshed and eliminate the need to configure it more complexly.